### PR TITLE
Cleanup OpArg, make immediates more explicit.

### PR DESCRIPTION
**Pull request by phire**
Instead of just casting OpArg::offset when needed, add some
accessor functions.

Also add some safety asserts to catch any mistakes.

### CODE REVIEW

**Comment by phire** [comment]
As requested by flacs on IRC, Made offset private. 

Now the only way to access offset externally is via IncrementAddress.

File: Source/Core/Common/x64Emitter.h [hunk 1]
```diff
@@ -188,10 +197,20 @@ struct OpArg
 		else
 			return INVALID_REG;
 	}
+
+	void AddMemOffset(int val)
+	{ 
+		_dbg_assert_msg_(DYNA_REC, scale == SCALE_RIP || (scale <= SCALE_ATREG && scale > SCALE_NONE),
+						 "Tried to increment an OpArg which doesn't have an offset");
```
Comment (Tilka) [inline Source/Core/Common/x64Emitter.h]: alignment

**Comment by Tilka** [comment]
lgtm

**Comment by Sintendo** [comment]
What if I need to get at the immediate value, but don't know/care what size was used for it?

**Comment by phire** [comment]
The size of the immediate is always known. For example, all immediates used by the powerpc jit are 32bit.

**Pull request merged**